sklego/model_selection.py

```diff
@@ -0,0 +1,44 @@
+from sklearn.model_selection._split import _BaseKFold
+from sklearn.cluster import k_means
+import numpy as np
+import pandas as pd
+import warnings
+
+
+class KMeansFold(_BaseKFold):
+    """
+    K-Means cross validator
+
+    :param n_splits: Number of splits (clusters) to fold on
+    :param kmeans_kwargs: Extra kwargs for k_means apart from n_clusters
+    """
+
+    def __init__(self, n_splits='default', k_means_kwargs=None):
```

Comment:
I would expect this class to use a `random_state` to get some reproducability of results

Reply:
Check, I have added it